crawl-ref/source/webserver/static/scripts/client.js

```diff
@@ -1099,6 +1099,8 @@ function (exports, $, key_conversion, chat, comm) {
         set("xl", data.xl);
         set("char", data.char);
         set("place", data.place);
+        set("turn", data.turn);
```

Comment:
This PR overall looks pretty good to me, thanks for the patch! The one caveat here is that for the initial merge it would be best to add a conditional here around these two lines: the python code will require a webtiles server restart, and the js code and the html template will update potentially right away (though `client.js` is dependent on the browser cache so that could take a while too, it's a bit unpredictable). Until a server restart happens (some servers restart very infrequently) this js code will produce an error. It's probably also best to use jquery to hide the columns if `data` doesn't have `turn` and `dur`, or maybe even start them hidden. Eventually this migration code could be removed once all servers have these changes.

Or another way to put this: I recommend testing this code with the python side of things commented out to make sure the js UI doesn't break during migration, which I think it is likely to right now.

Reply:
Good point, I hadn't considered that.